frontend/packages/local-storage-operator-plugin/src/components/local-volume-set/local-volume-set-inner.tsx

```diff
@@ -0,0 +1,220 @@
+import * as React from 'react';
+import {
+  FormGroup,
+  TextInput,
+  Radio,
+  Expandable,
+  TextInputTypes,
+  Text,
+  TextVariants,
+} from '@patternfly/react-core';
+import { Dropdown } from '@console/internal/components/utils';
+import { ListPage } from '@console/internal/components/factory';
+import { k8sList, NodeKind } from '@console/internal/module/k8s';
+import { getName } from '@console/shared';
+import { NodeModel } from '@console/internal/models';
+import { hasTaints } from '@console/ceph-storage-plugin/src/utils/install';
+import { NodesSelectionList } from './nodes-selection-list';
+import {
+  State,
+  Action,
+  diskModeDropdownItems,
+  diskTypeDropdownItems,
+  diskSizeUnitOptions,
+} from './state';
+import './create-local-volume-set.scss';
+
+let allNodeNames: string[] = [];
+
+export const LocalVolumeSetInner: React.FC<LocalVolumeSetInnerProps> = (props) => {
+  const { dispatch, state, match } = props;
+  const { ns, appName } = match;
+
+  React.useEffect(() => {
+    k8sList(NodeModel)
+      .then((nodes: NodeKind[]) => {
+        allNodeNames = nodes.filter((node) => !hasTaints(node)).map((node) => getName(node));
+        dispatch({ type: 'setNodeNames', value: allNodeNames });
+      })
+      .catch(() => dispatch({ type: 'setNodeNames', value: [] }));
+  }, [ns, appName, dispatch]);
+
+  React.useEffect(() => {
+    if (!state.showNodesList) {
+      dispatch({ type: 'setNodeNames', value: allNodeNames });
```

Comment:
This is required when user selects `All Nodes` where the `ListPage` component will not be rendered.

Reply:
Then we can dispatch this action when user selects `All nodes`.

Reply:
1) This would not change when `allNodeNames` changes. When `allNodesNames` value changes this function needs to be called. This can be a source of an indeterministic bug where the state is set correctly only if the fetch happens before the `state.showNodesList` changes. 
2) Looks like maintaining a set of Nodes in the state would eliminate the need for this useEffect code block altogether.